tsdb/wal/wal.go

```diff
@@ -335,6 +342,11 @@ func (w *WAL) Repair(origErr error) error {
 	// But that's not generally applicable if the records have any kind of causality.
 	// Maybe as an extra mode in the future if mid-WAL corruptions become
 	// a frequent concern.
+
+	if origErr.Error() == ErrorUnsequentialSegments {
+		return w.repairUnsequentialSegments()
+	}
+
 	err := errors.Cause(origErr) // So that we can pick up errors even if wrapped.
 
 	cerr, ok := err.(*CorruptionErr)
```

Comment:
now that the Repair handles different errors I would move everything after this in its own func